### PR TITLE
Add Tasks.org support

### PR DESCRIPTION
**Pull request by abaker**

### CODE REVIEW

**Comment by tasn** [comment]
Note to self: make sure to test the import functionality.

File: app/src/main/java/com/etesync/syncadapter/syncadapter/RequestSync.kt [hunk 1]
```diff
@@ -4,12 +4,13 @@ import android.accounts.Account
...
 import android.os.Bundle
 import android.provider.CalendarContract
-import at.bitfire.ical4android.TaskProvider
+import at.bitfire.ical4android.TaskProvider.Companion.OPENTASK_PROVIDERS
 import com.etesync.syncadapter.App
 
 
 fun requestSync(account: Account?) {
-    val authorities = arrayOf(App.addressBooksAuthority, CalendarContract.AUTHORITY, TaskProvider.ProviderName.OpenTasks.authority)
+    val authorities = arrayOf(App.addressBooksAuthority, CalendarContract.AUTHORITY) +
```
Comment (tasn) [inline app/src/main/java/com/etesync/syncadapter/syncadapter/RequestSync.kt:12]: Should use `getWantedTaskSyncProvider`

File: app/src/main/java/com/etesync/syncadapter/ui/CreateCollectionFragment.kt [hunk 2]
```diff
@@ -84,15 +85,14 @@ class CreateCollectionFragment : DialogFragment(), LoaderManager.LoaderCallbacks
...
             try {
-                var authority: String = ""
-
                 val data = (context.applicationContext as App).data
 
                 // 1. find service ID
-                when (info.enumType){
-                    CollectionInfo.Type.ADDRESS_BOOK -> authority = App.addressBooksAuthority
-                    CollectionInfo.Type.CALENDAR -> authority = CalendarContract.AUTHORITY
-                    CollectionInfo.Type.TASKS -> authority = TaskProvider.ProviderName.OpenTasks.authority
```
Comment (tasn) [inline app/src/main/java/com/etesync/syncadapter/ui/CreateCollectionFragment.kt:95]: Should use `getWantedTaskSyncProvider`

File: app/src/main/java/com/etesync/syncadapter/ui/JournalItemActivity.kt [hunk 3]
```diff
@@ -116,15 +117,17 @@ class JournalItemActivity : BaseActivity(), Refreshable {
...
-                val provider = TaskProvider.acquire(this, TaskProvider.ProviderName.OpenTasks)!!
-                val localTaskList = LocalTaskList.findByName(account, provider, LocalTaskList.Factory, info.uid!!)!!
-                val task = Task.tasksFromReader(StringReader(syncEntry.content))[0]
-                var localTask = localTaskList.findByUid(task.uid!!)
-                if (localTask != null) {
-                    localTask.updateAsDirty(task)
-                } else {
-                    localTask = LocalTask(localTaskList, task, task.uid, null)
-                    localTask.addAsDirty()
+                OPENTASK_PROVIDERS.forEach {
```
Comment (tasn) [inline app/src/main/java/com/etesync/syncadapter/ui/JournalItemActivity.kt:120]: I think this should use `getWantedTaskSyncProvider`

File: app/src/main/java/com/etesync/syncadapter/ui/ViewCollectionActivity.kt [hunk 4]
```diff
@@ -212,7 +213,9 @@ class ViewCollectionActivity : BaseActivity(), Refreshable {
                 }
                 CollectionInfo.Type.TASKS -> {
                     try {
-                        val providerClient = TaskProvider.acquire(this@ViewCollectionActivity, TaskProvider.ProviderName.OpenTasks)
+                        val providerClient = OPENTASK_PROVIDERS.mapNotNull {
+                            TaskProvider.acquire(this@ViewCollectionActivity, it)
```
Comment (tasn) [inline app/src/main/java/com/etesync/syncadapter/ui/ViewCollectionActivity.kt:217]: Should use `getWantedTaskSyncProvider`

File: app/src/main/java/com/etesync/syncadapter/ui/importlocal/ImportFragment.kt [hunk 5]
```diff
@@ -282,40 +283,44 @@ class ImportFragment : DialogFragment() {
 
                     finishParsingFile(tasks.size)
 
-                    val provider = TaskProvider.acquire(context, TaskProvider.ProviderName.OpenTasks)
-                    if (provider == null) {
-                        result.e = Exception("Failed to acquire tasks content provider.")
-                        return result
+                    val providers = OPENTASK_PROVIDERS.mapNotNull {
```
Comment (tasn) [inline app/src/main/java/com/etesync/syncadapter/ui/importlocal/ImportFragment.kt:286]: Should use `getWantedTaskSyncProvider`

**Comment by abaker** [comment]
I made the requested changes. Let me know if you think I need to make that change anywhere else

**Comment by tasn** [comment]
Oh FFS, I hate how Github doesn't detect merges from the command line!!!

Anyhow, merged, sorry I have to close it manually.

**Pull request closed without merging**